core/src/mender-client.c

```diff
@@ -1272,10 +1117,13 @@ mender_client_update_work_function(void) {
         cJSON_Delete(mender_client_deployment_data);
         mender_client_deployment_data = NULL;
     }
-    mender_artifact_release_ctx(mender_artifact_ctx);
+    mender_log_info("Releasing artifact context");
+    // we will be restarting anyways
```

Comment:
Maybe you can consider adding a `TODO` comment that we should check the context releasing (deinitialization; cleanup) before the restart.